cmd.sh

```diff
@@ -2,91 +2,4 @@
-
-# figure out whether SSL is enabled in user_config.h
-if grep -Eq "^#define CLIENT_SSL_ENABLE" app/include/user_config.h; then
-    SSL="true"
-else
-    SSL="false"
-fi
-
-# check whether the user made changes to the version file, if so then assume she wants to use a custom version rather
-# than the one that would be set here -> we can't modify it
-if git diff --quiet app/include/user_version.h; then
-    CAN_MODIFY_VERSION=true
-else
-    CAN_MODIFY_VERSION=false
-fi
-
-# use the Git branch and the current time stamp to define image name if IMAGE_NAME not set
-if [ -z "$IMAGE_NAME" ]; then
-  IMAGE_NAME=${BRANCH}_"$(date "+%Y%m%d-%H%M")"
-else
-  true
-fi
-
-# unpack esp-open-sdk.tar.gz into a directory that is NOT the bound mount directory (i.e. inside the Docker image)
-export HOME=/tmp/esp
-rm -rf ${HOME}
-mkdir ${HOME}
-
-cp tools/esp-open-sdk.tar.* ${HOME}
-cd ${HOME}
-# support older build chains (before we re-packaged it)
-if [ -f ./esp-open-sdk.tar.xz ]; then
-  tar -Jxvf esp-open-sdk.tar.xz
-else
-  tar -zxvf esp-open-sdk.tar.gz
-fi
-
-export PATH=$PATH:$PWD/esp-open-sdk/sdk:$PWD/esp-open-sdk/xtensa-lx106-elf/bin
-export CCACHE_DIR=/opt/nodemcu-firmware/.ccache
-
-cd /opt/nodemcu-firmware
-
-# modify user_version.h to provide more info in NodeMCU welcome message, doing this by passing
-# EXTRA_CCFLAGS="-DBUILD_DATE=... AND -DNODE_VERSION=..." to make turned into an escaping/expanding nightmare for which
-# I never found a good solution
-if [ "$CAN_MODIFY_VERSION" = true ]; then
-  MODULES=$(awk '/^[ \t]*#define LUA_USE_MODULES/{modules=modules sep tolower(substr($2,17));sep=","}END{if(length(modules)==0)modules="-";print modules}' app/include/user_modules.h)
-  sed -i '/#define NODE_VERSION[[:space:]]/ s/$/ " built with Docker provided by frightanic.com\\n\\tbranch: '"$BRANCH"'\\n\\tcommit: '"$COMMIT_ID"'\\n\\tSSL: '"$SSL"'\\n\\tmodules: '"$MODULES"'\\n"/g' app/include/user_version.h
-  sed -i 's/"unspecified"/"created on '"$BUILD_DATE"'\\n"/g' app/include/user_version.h
-fi
-
-# make a float build if !only-integer
-if [ -z "$INTEGER_ONLY" ]; then
-  make WRAPCC="$(which ccache)" clean all
-  cd bin
-  srec_cat -output nodemcu_float_"${IMAGE_NAME}".bin -binary 0x00000.bin -binary -fill 0xff 0x00000 0x10000 0x10000.bin -binary -offset 0x10000
-  # copy and rename the mapfile to bin/
-  cp ../app/mapfile nodemcu_float_"${IMAGE_NAME}".map
-  cd ../
-else
-  true
-fi
-
-# make an integer build
-if [ -z "$FLOAT_ONLY" ]; then
-  make WRAPCC="$(which ccache)" EXTRA_CCFLAGS="-DLUA_NUMBER_INTEGRAL" clean all
-  cd bin
-  srec_cat -output nodemcu_integer_"${IMAGE_NAME}".bin -binary 0x00000.bin -binary -fill 0xff 0x00000 0x10000 0x10000.bin -binary -offset 0x10000
-  # copy and rename the mapfile to bin/
-  cp ../app/mapfile nodemcu_integer_"${IMAGE_NAME}".map
-  cd ../
-else
-  true
-fi
-
-# revert the changes made to the version params
-if [ "$CAN_MODIFY_VERSION" = true ]; then
-  git checkout app/include/user_version.h
-fi
+cat ../read.me
```

Comment:
Great you introduced the readme! However, I'm reluctant to change the command and thus give up backwards compatibility. Every script that's calling the image will break.

It may very well be a matter of personal taste and style but I'm currently leaning towards passing a param for the LFS stuff. I do acknowledge though that your proposal has its charm as it's much more  Dockerish to treat the two operations as explicit command.

Reply:
hmm, also could leave the original functionality in place and only have an extra command to create an LFS image.
Not sure whether it makes sense to print the read.me at the end of the build then.

BTW: the build script has a weakness in building Float only builds. When int builds are configured in the config file it still builds an integer and names it float.

Reply:
I'll sleep over that design decision.

> When int builds are configured in the config file it still builds an integer and names it float.

That recently surfaced somewhere in a comment. We should get rid of those INTEGER_ONLY/FLOAT_ONLY params and only honour the configuration in the `.h` file.

Reply:
Sounds good :-)

Reply:
Will add the new functionality in the next days. The developer will want to tweak the list of modules anyway, so why not adapt integer and LFS settings as well.

Reply:
Just coming back to this and wondering if you'll have anything ready soon? Would be a pity if we both spent time for the same stuff. Want to get this done and over with asap so we can proceed with LFS merging in NodeMCU.

Reply:
It is ready except for documentation and some testing.
Will be ready in about 2 Days.
Hope that is early enough so you can wait.

Reply:
Great, looking forward, thanks.

Reply:
@marcelstoer please have another look.